packages/jsii-pacmak/lib/targets/go/package.ts

```diff
@@ -114,16 +122,63 @@ export abstract class Package {
     code.closeFile(this.file);
 
     this.emitGoInitFunction(context);
+    this.emitSubmodules(context);
 
-    this.emitInternalPackages(context);
+    this.emitInternal(context);
   }
 
-  public emitInternalPackages(context: EmitContext) {
+  public emitSubmodules(context: EmitContext) {
     for (const submodule of this.submodules) {
       submodule.emit(context);
     }
   }
 
+  /**
+   * Determines if `type` comes from a foreign package.
+   */
+  public isExternalType(type: GoClass | GoInterface) {
+    return type.pkg !== this;
+  }
+
+  /**
+   * Returns the name of the embed field used to embed a base class/interface in a
+   * struct.
+   *
+   * @returns If the base is in the same package, returns the proxy name of the
+   * base under `embed`, otherwise returns a unique symbol under `embed` and the
+   * original interface reference under `original`.
+   *
+   * @param type The base type we want to embed
+   */
+  public resolveEmbeddedType(type: GoClass | GoInterface): EmbeddedType {
+    if (!this.isExternalType(type)) {
+      return {
+        embed: type.proxyName,
+        fieldName: type.proxyName,
+      };
+    }
+
+    const exists = this.embeddedTypes[type.fqn];
+    if (exists) {
+      return exists;
+    }
+
+    const typeref = new GoTypeRef(this.root, type.type.reference);
+    const original = typeref.scopedInterfaceName(this);
+
+    const slug = original.replace(/[^A-Za-z0-9]/g, '');
+    const aliasName = `Type_${slug}`;
```

Comment:
Suggest using TWO `_` delimiter, so it will _NEVER_ conflict with a nested type (those would have only 1).

Reply:
Fixed